### PR TITLE
Add vision-language model pretraining strategies blog

### PR DESCRIPTION
**Pull request by alaradirik**
This blog is meant to give a non-exhaustive overview of vision-language models and pre-training strategies.

Sections:
1. Introduction
2. Learning Strategies
    1. Contrastive Learning
    2. PrefixLM
    3. Multimodal Fusing with Cross Attention
    4. MLM / ITM
    5. No Training
3. Datasets
4. Supporting Vision-Language Models in 🤗 Transformers
5. Emerging Areas of Research
6. Conclusion

- [x] Add thumbnail

### CODE REVIEW

File: vision_language_pretraining.md [hunk 1]
```diff
@@ -0,0 +1,390 @@
+---
+title: "A Dive into Pretraining Strategies for Vision-Language Models"
```
Comment (sayakpaul) [inline vision_language_pretraining.md:2]: Maybe just Vision-Language Models?

File: vision_language_pretraining.md [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+thumbnail: /blog//assets/128_vision_language_pretraining/thumbnail.png
+---
+
+<h1>A Dive into Pretraining Strategies for Vision-Language Models</h1>
+
+<div class="blog-metadata">
+    <small>Published January 27, 2023.</small>
```
Comment (sayakpaul) [inline vision_language_pretraining.md:9]: What about publishing it later next week? We have an upcoming blog on "CV at 🤗" planned.

File: vision_language_pretraining.md [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+2. [Vision-Language Models: Learning Strategies]
+    1. [Contrastive Learning]
+    2. [PrefixLM]
+    3. [Multimodal Fusing with Cross Attention]
+    4. [MLM / ITM]
+    5. [No Training]
+3. [Vision-Language Models: Datasets]
+4. [Supporting Vision-Language Models in 🤗 Transformers]
+5. [Emerging Areas of Research]
+6. [Conclusion]
```
Comment (sayakpaul) [inline vision_language_pretraining.md:51]: Update with section hyperlinks?

File: vision_language_pretraining.md [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+
+But these inputs and outputs can take several forms. Below we give some examples:
+-   Image retrieval from natural language text.
+-   Phrase grounding, i.e., performing object detection from an input image and natural language phrase (example: A **young person** swings a **bat**).
+-   Visual question answering, i.e., finding answers from an input image and a question in natural language.
+-   Generate a caption for a given image. This can also take the form of conditional text generation, where you'd start with a natural language prompt and an image.
+-   Detection of hate speech from social media content involving both images and text modalities.
+
+
+# Vision-Language Models: Learning Strategies
```
Comment (sayakpaul) [inline vision_language_pretraining.md:77]: Leave a gap between the header and the text?

File: vision_language_pretraining.md [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+
+Let’s start with ViLT and download a model pre-trained on the VQA dataset. We can do this by simply initializing the corresponding model class and calling the `from_pretrained()` method to download our desired checkpoint.
+
+```py
+
+from transformers import ViltProcessor, ViltForQuestionAnswering
+
+model = ViltForQuestionAnswering.from_pretrained("dandelin/vilt-b32-finetuned-vqa")
+
+```
```
Comment (sayakpaul) [inline vision_language_pretraining.md:245]: Can we remove the unnecessary spaces?

File: vision_language_pretraining.md [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+
+image = Image.open(requests.get(url, stream=True).raw)
+
+text = "How many cats are there?"
+
+# prepare inputs
+
+inputs = processor(image, text, return_tensors="pt")
+
+```
```
Comment (sayakpaul) [inline vision_language_pretraining.md:269]: Can we remove the unnecessary spaces

File: vision_language_pretraining.md [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+
+    outputs = model(**inputs)
+
+logits = outputs.logits
+
+idx = logits.argmax(-1).item()
+
+print("Predicted answer:", model.config.id2label[idx])
+
+```
```
Comment (sayakpaul) [inline vision_language_pretraining.md:289]: Can we remove the unnecessary spaces?

File: vision_language_pretraining.md [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+
+```py
+
+from transformers import CLIPSegProcessor, CLIPSegForImageSegmentation
+
+processor = CLIPSegProcessor.from_pretrained("CIDAS/clipseg-rd64-refined")
+
+model = CLIPSegForImageSegmentation.from_pretrained("CIDAS/clipseg-rd64-refined")
+
+```
```
Comment (sayakpaul) [inline vision_language_pretraining.md:306]: Can we remove the unnecessary spaces?

File: vision_language_pretraining.md [hunk 9]
```diff
@@ -0,0 +1,390 @@
...
+
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+
+image = Image.open(requests.get(url, stream=True).raw)
+
+texts = ["a cat", "a remote", "a blanket"]
+
+inputs = processor(text=texts, images=[image] * len(texts), padding=True, return_tensors="pt")
+
+```
```
Comment (sayakpaul) [inline vision_language_pretraining.md:324]: Can we remove the unnecessary spaces?

File: vision_language_pretraining.md [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+
+    outputs = model(**inputs)
+
+logits = outputs.logits
+
+print(logits.shape)
+
+>>> torch.Size([3, 352, 352])
+
+```
```
Comment (sayakpaul) [inline vision_language_pretraining.md:342]: Same.

File: vision_language_pretraining.md [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+
+[a.axis('off') for a in ax.flatten()]
+
+ax[0].imshow(image)
+
+[ax[i+1].imshow(torch.sigmoid(logits[i][0])) for i in range(len(texts))];
+
+[ax[i+1].text(0, -15, prompt) for i, prompt in enumerate(texts)]
+
+```
```
Comment (sayakpaul) [inline vision_language_pretraining.md:362]: Same.

File: vision_language_pretraining.md [hunk 12]
```diff
@@ -0,0 +1,390 @@
...
+
+
+# Conclusion
+
+There have been incredible advances in multimodal models in recent years, with vision-language models making the biggest leap both in terms of performance and the variety of use cases and applications. In this blog, we talked about the latest advancements in vision-language models, as well as what multimodal datasets are available and which pre-training strategies we can use to train and fine-tune such models. We also showed how these models are integrated into 🤗 Transformers and how you can use them to perform various tasks with a few lines of code. 
+
+We are continuing to integrate the most impactful computer vision and multimodal models and would love to hear back from you. To stay up to date with the latest news in multimodal research, you can follow us on Twitter: [@adirik](https://twitter.com/adirik), [@NielsRogge](https://twitter.com/NielsRogge), [@apsdehal](https://twitter.com/apsdehal), [@a_e_roberts](https://twitter.com/a_e_roberts),  @RisingSayak](https://mobile.twitter.com/a_e_roberts), and [@huggingface](https://twitter.com/huggingface).
+
```
Comment (sayakpaul) [inline vision_language_pretraining.md:388]: Looks like a "[" is missing from my username.

**Review by sayakpaul — APPROVED** [review]
Thanks a lot! Left a few minor nits. 

Also, did you preview the blog using [codespaces](https://github.com/huggingface/moon-landing#codespace)? Also, we need to update the `_blog.yml`. 

@apsdehal @amyeroberts FYI. 

Also Cc: @NielsRogge @patil-suraj @younesbelkada

File: vision_language_pretraining.md (hunk 2 above)
Comment (alaradirik) [reply vision_language_pretraining.md:9]: Makes sense, would February 2nd or 3rd work?

File: vision_language_pretraining.md (hunk 3 above)
Comment (alaradirik) [reply vision_language_pretraining.md:51]: Yes, updating it :)

File: vision_language_pretraining.md (hunk 1 above)
Comment (alaradirik) [reply vision_language_pretraining.md:2]: Makes sense, how about "A Deep into for Vision-Language Models"?

File: vision_language_pretraining.md (hunk 2 above)
Comment (sayakpaul) [reply vision_language_pretraining.md]: Let's do it on 3rd?

File: vision_language_pretraining.md (hunk 1 above)
Comment (sayakpaul) [reply vision_language_pretraining.md:2]: "Deep into" seems incomplete to me.

Comment (sayakpaul) [reply vision_language_pretraining.md:2]: Or maybe just "A Dive into Vision-Language Models"?

**Comment by sayakpaul** [comment]
@alaradirik I had to update an entry for the blog in `_blog.yml` to preview using Codespaces. 

Some things that came up:

1) The hyperlinks are not rendering in the image captions. 

Screenshot 2023-01-28 at 7 57 00 AM

2) The hyperlinks are not rendering here: 

Screenshot 2023-01-28 at 7 56 03 AM

**Review by younesbelkada — APPROVED** [review]
Amazing blogpost 🔥 ! 
BLIP also performs ITM, but not sure about its training strategy so I am not sure if it's relevant to add it in `4) Masked-Language Modeling / Image-Text Matching` wdyt?

File: vision_language_pretraining.md (hunk 1 above)
Comment (alaradirik) [reply vision_language_pretraining.md:2]: Sorry, I meant "A Deep Dive into Vision-Language Models" but "A Dive into Vision-Language Models" also works

Comment (sayakpaul) [reply vision_language_pretraining.md:2]: Alright! Let's update with the latter one the.  I am outside currently, if you can, could you do it if possible?

File: vision_language_pretraining.md [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+In this section, we’ll discuss some of the typical pre-training objectives and strategies for vision-language models that have been shown to perform well as far as their transfer performance is concerned. We’ll also touch upon additional interesting things that are either specific to these objectives or can be used as general components for pre-training.
```
Comment (osanseviero) [inline vision_language_pretraining.md:80]:
```suggestion
In this section, we'll discuss some of the typical pre-training objectives and strategies for vision-language models that have been shown to perform well regarding their transfer performance. We'll also touch upon additional interesting things that are either specific to these objectives or can be used as general components for pre-training. 
```

File: vision_language_pretraining.md [hunk 14]
```diff
@@ -0,0 +1,326 @@
...
+
+Models finetuned on the question-answering downstream task, such as [ViLT](https://arxiv.org/abs/2102.03334) and [GLIP](https://arxiv.org/abs/2112.03857), most commonly use the [VQA](https://visualqa.org/) (visual question-answering), [VQA v2](https://visualqa.org/), [NLVR2](https://lil.nlp.cornell.edu/nlvr/), [OKVQA](https://okvqa.allenai.org/), [TextVQA](https://huggingface.co/datasets/textvqa), [TextCaps](https://textvqa.org/textcaps/) and [VizWiz](https://vizwiz.org/) datasets. These datasets typically contain images paired with multiple open-ended questions and answers. Furthermore, datasets such as VizWiz and TextCaps can also be used for image segmentation and object localization downstream tasks. Some other interesting multi-modal downstream datasets are [Hateful Memes](https://huggingface.co/datasets/limjiayi/hateful_memes_expanded) for multi-modal classification, [SNLI-VE](https://github.com/necla-ml/SNLI-VE) for visual entailment prediction, and [Winoground](https://huggingface.co/datasets/facebook/winoground) for visio-linguistic compositional reasoning.
```
Comment (osanseviero) [inline vision_language_pretraining.md:183]:
```suggestion
Models fine-tuned on the question-answering downstream task, such as [ViLT](https://arxiv.org/abs/2102.03334) and [GLIP](https://arxiv.org/abs/2112.03857), most commonly use the [VQA](https://visualqa.org/) (visual question-answering), [VQA v2](https://visualqa.org/), [NLVR2](https://lil.nlp.cornell.edu/nlvr/), [OKVQA](https://okvqa.allenai.org/), [TextVQA](https://huggingface.co/datasets/textvqa), [TextCaps](https://textvqa.org/textcaps/) and [VizWiz](https://vizwiz.org/) datasets. These datasets typically contain images paired with multiple open-ended questions and answers. Furthermore, datasets such as VizWiz and TextCaps can also be used for image segmentation and object localization downstream tasks. Some other interesting multi-modal downstream datasets are [Hateful Memes](https://huggingface.co/datasets/limjiayi/hateful_memes_expanded) for multi-modal classification, [SNLI-VE](https://github.com/necla-ml/SNLI-VE) for visual entailment prediction, and [Winoground](https://huggingface.co/datasets/facebook/winoground) for visio-linguistic compositional reasoning. 
```

File: vision_language_pretraining.md [hunk 15]
```diff
@@ -0,0 +1,326 @@
...
+    <em>Frozen PrefixLM pre-training strategy (<a href=https://lilianweng.github.io/posts/2022-06-09-vlm>image source</a>)</em>
+</p>
+
+While fusing visual information into a language model is highly effective, being able to use a pre-trained language model (LM) without the need for fine-tuning would be much more efficient. Hence, another pre-training objective in vision-language models is learning image embeddings that are aligned with a frozen language model. 
+
+Models such as [Frozen](https://arxiv.org/abs/2106.13884), [MAPL, ](https://arxiv.org/abs/2210.07179)and [ClipCap](https://arxiv.org/abs/2111.09734) use this Frozen PrefixLM pre-training objective. They only update the parameters of the image encoder during training to generate image embeddings that can be used as a prefix to the pre-trained, frozen language model, in a similar fashion to the PrefixLM objective discussed above. Both Frozen and ClipCap are trained on aligned image-text (caption) datasets with the objective of generating the next token in the caption, given the image embeddings and the prefix text.
```
Comment (osanseviero) [inline vision_language_pretraining.md:128]:
```suggestion
Models such as [Frozen](https://arxiv.org/abs/2106.13884), [MAPL, ](https://arxiv.org/abs/2210.07179)and [ClipCap](https://arxiv.org/abs/2111.09734) use this Frozen PrefixLM pre-training objective. They only update the parameters of the image encoder during training to generate image embeddings that can be used as a prefix to the pre-trained, frozen language model in a similar fashion to the PrefixLM objective discussed above. Both Frozen and ClipCap are trained on aligned image-text (caption) datasets with the objective of generating the next token in the caption, given the image embeddings and the prefix text. 
```

File: vision_language_pretraining.md [hunk 16]
```diff
@@ -0,0 +1,326 @@
...
+
+Models such as VisualGPT use a visual encoder to embed images and feed the visual embeddings to the cross-attention layers of a pre-trained language decoder module to generate plausible captions. A more recent work, [FIBER](http://arxiv.org/abs/2206.07643), inserts cross-attention layers with a gating mechanism into both vision and language backbones for more efficient multimodal fusing and enables various other downstream tasks such as image-text retrieval and open vocabulary object detection.
```
Comment (osanseviero) [inline vision_language_pretraining.md:142]:
```suggestion
Models such as VisualGPT use a visual encoder to embed images and feed the visual embeddings to the cross-attention layers of a pre-trained language decoder module to generate plausible captions. A more recent work, [FIBER](http://arxiv.org/abs/2206.07643), inserts cross-attention layers with a gating mechanism into both vision and language backbones, for more efficient multimodal fusing and enables various other downstream tasks, such as image-text retrieval and open vocabulary object detection. 

```

File: vision_language_pretraining.md [hunk 17]
```diff
@@ -0,0 +1,326 @@
...
+
+Models such as VisualGPT use a visual encoder to embed images and feed the visual embeddings to the cross-attention layers of a pre-trained language decoder module to generate plausible captions. A more recent work, [FIBER](http://arxiv.org/abs/2206.07643), inserts cross-attention layers with a gating mechanism into both vision and language backbones for more efficient multimodal fusing and enables various other downstream tasks such as image-text retrieval and open vocabulary object detection. 
+### 4) Masked-Language Modeling / Image-Text Matching
+
+Another line of vision-language models use a combination of Masked-Language Modeling (MLM) and Image-Text Matching (ITM) objectives to align specific parts of images with text and enable various downstream tasks such as visual question answering, visual commonsense reasoning, text-based image retrieval and text-guided object detection. Models that follow this pre-training setup include  [VisualBERT](https://arxiv.org/abs/1908.03557), [FLAVA](https://arxiv.org/abs/2112.04482), [ViLBERT](https://arxiv.org/abs/1908.02265) and [LXMERT](https://arxiv.org/abs/1908.07490).
```
Comment (osanseviero) [inline vision_language_pretraining.md:145]:
```suggestion
Another line of vision-language models uses a combination of Masked-Language Modeling (MLM) and Image-Text Matching (ITM) objectives to align specific parts of images with text and enable various downstream tasks such as visual question answering, visual commonsense reasoning, text-based image retrieval, and text-guided object detection. Models that follow this pre-training setup include [VisualBERT](https://arxiv.org/abs/1908.03557), [FLAVA](https://arxiv.org/abs/2112.04482), [ViLBERT](https://arxiv.org/abs/1908.02265) and [LXMERT](https://arxiv.org/abs/1908.07490).
```

File: vision_language_pretraining.md [hunk 18]
```diff
@@ -0,0 +1,326 @@
...
+
+Another work, FLAVA, consists of an image encoder, a text encoder, and a multimodal encoder to fuse and align the image and text representations for multimodal reasoning, all of which are based on transformers. In order to achieve this, FLAVA uses a variety of pre-training objectives: MLM, ITM, as well as Masked-Image Modeling (MIM), and contrastive learning.
+### 5) No Training 
+
+Finally, there are various optimization strategies that aim to bridge image and text representations using the pre-trained image and text models or adapt pre-trained multimodal models to new downstream tasks without any additional training.
```
Comment (osanseviero) [inline vision_language_pretraining.md:159]:
```suggestion
Finally, various optimization strategies aim to bridge image and text representations using the pre-trained image and text models or adapt pre-trained multimodal models to new downstream tasks without additional training. 
```

File: vision_language_pretraining.md [hunk 19]
```diff
@@ -0,0 +1,326 @@
...
+    <em>Crafting a similarity search space using pre-trained, frozen unimodal image and text encoders (<a href=https://luca.moschella.dev/publication/norelli-asif-2022>image source</a>)</em>
+</p>
+
+
+[ASIF](https://arxiv.org/abs/2210.01738) proposes a simple method to turn pre-trained uni-modal image and text models into a multimodal model for image captioning using a relatively small multimodal dataset without additional training. The key intuition behind ASIF is that captions of similar images are also similar to each other. Hence we can perform a similarity-based search by crafting a relative representation space using a small dataset of ground-truth multimodal pairs.
+
+## Vision-Language Models: Datasets
+
+Vision-language models are typically trained on large image and text datasets with different structures based on the pre-training objective. After they are pre-trained, they are further fine-tuned on various downstream tasks using task-specific datasets. In this section, we provide an overview of some popular pre-training and downstream datasets used for training and evaluating vision-language models.
```
Comment (osanseviero) [inline vision_language_pretraining.md:173]:
```suggestion
Vision-language models are typically trained on large image and text datasets with different structures based on the pre-training objective. After they are pre-trained, they are further fine-tuned on various downstream tasks using task-specific datasets. This section provides an overview of some popular pre-training and downstream datasets used for training and evaluating vision-language models.

```

File: vision_language_pretraining.md [hunk 20]
```diff
@@ -0,0 +1,326 @@
...
+
+Note that even image-text datasets that solely consist of human-generated captions such as Flickr30K are inherently noisy as users do not always write descriptive or reflective captions for their images. To overcome this issue, datasets such as the LAION-5B dataset leverage CLIP or other pre-trained multimodal models to filter noisy data and create high-quality multimodal datasets. Furthermore, some vision-language models such as ALIGN, propose further preprocessing steps and create their own high-quality datasets. Other vision-language datasets, such as the [LSVTD](https://davar-lab.github.io/dataset/lsvtd.html) and [WebVid](https://github.com/m-bain/webvid) datasets, consist of video and text modalities, although at a smaller scale.
```
Comment (osanseviero) [inline vision_language_pretraining.md:178]:
```suggestion
Even image-text datasets consisting solely of human-generated captions, such as Flickr30K, are inherently noisy as users only sometimes write descriptive or reflective captions for their images. To overcome this issue, datasets such as the LAION-5B dataset leverage CLIP or other pre-trained multimodal models to filter noisy data and create high-quality multimodal datasets. Furthermore, some vision-language models, such as ALIGN, propose further preprocessing steps and create their own high-quality datasets. Other vision-language datasets, such as the [LSVTD](https://davar-lab.github.io/dataset/lsvtd.html) and [WebVid](https://github.com/m-bain/webvid) datasets, consist of video and text modalities, although at a smaller scale.

```

File: vision_language_pretraining.md [hunk 21]
```diff
@@ -0,0 +1,326 @@
...
+
+Note that vision-language models are used for various classical NLP and computer vision tasks such as text or image classification, and typically use uni-modal datasets ([SST2](https://huggingface.co/datasets/sst2), [ImageNet-1k](https://huggingface.co/datasets/imagenet-1k), for example) for such downstream tasks. In addition, datasets such as [COCO](https://cocodataset.org/), and [Conceptual Captions](https://ai.google.com/research/ConceptualCaptions/) are commonly used both in the pre-training of models and also for the caption generation downstream task.
```
Comment (osanseviero) [inline vision_language_pretraining.md:185]:
```suggestion
Note that vision-language models are used for various classical NLP and computer vision tasks such as text or image classification and typically use uni-modal datasets ([SST2](https://huggingface.co/datasets/sst2), [ImageNet-1k](https://huggingface.co/datasets/imagenet-1k), for example) for such downstream tasks. In addition, datasets such as [COCO](https://cocodataset.org/) and [Conceptual Captions](https://ai.google.com/research/ConceptualCaptions/) are commonly used both in the pre-training of models and also for the caption generation downstream task. 

```

File: vision_language_pretraining.md [hunk 22]
```diff
@@ -0,0 +1,326 @@
...
+
+Unlike other models, the `VisionEncoderDecoderModel` is a cookie-cutter model that can be used to initialize an image-to-text model with any pre-trained Transformer-based vision model as the encoder (e.g. ViT, BEiT, DeiT, Swin) and any pre-trained language model as the decoder (e.g. RoBERTa, GPT2, BERT, DistilBERT). In fact, TrOCR is an instance of this cookie-cutter class.
+
+Let’s go ahead and experiment with some of these models. We will use [ViLT](https://huggingface.co/docs/transformers/model_doc/vilt) for visual question answering and [CLIPSeg](https://huggingface.co/docs/transformers/model_doc/clipseg) for zero-shot image segmentation. First, let’s install 🤗Transformers: `pip install transformers`.
```
Comment (osanseviero) [inline vision_language_pretraining.md:211]:
```suggestion
Let’s go ahead and experiment with some of these models. We will use [ViLT](https://huggingface.co/docs/transformers/model_doc/vilt) for visual question answering and [CLIPSeg](https://huggingface.co/docs/transformers/model_doc/clipseg) for zero-shot image segmentation. First, let’s install 🤗Transformers: `pip install transformers`.

```

File: vision_language_pretraining.md [hunk 23]
```diff
@@ -0,0 +1,326 @@
...
+with torch.no_grad():
+    outputs = model(**inputs)
+
+logits = outputs.logits
+idx = logits.argmax(-1).item()
+print("Predicted answer:", model.config.id2label[idx])
+```
+
+Straight-forward, right? Let’s do another demonstration with CLIPSeg and see how we can perform zero-shot image segmentation with a few lines of code. 
+### CLIPSeg for zero-shot image segmentation
```
Comment (osanseviero) [inline vision_language_pretraining.md:254]:
```suggestion
Straight-forward, right? Let’s do another demonstration with CLIPSeg and see how we can perform zero-shot image segmentation with a few lines of code. 

### CLIPSeg for zero-shot image segmentation
```

File: vision_language_pretraining.md [hunk 24]
```diff
@@ -0,0 +1,326 @@
...
+import requests
+
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+image = Image.open(requests.get(url, stream=True).raw)
+texts = ["a cat", "a remote", "a blanket"]
+
+inputs = processor(text=texts, images=[image] * len(texts), padding=True, return_tensors="pt")
+```
+
+Similar to ViLT, it’s important to refer to the [original work](https://arxiv.org/abs/2112.10003) to see what kind of text prompts are used to train the model in order to get the best performance during inference. While CLIPSeg is trained on simple object descriptions (e.g., “a car”), its CLIP backbone is pre-trained on engineered text templates (e.g., “an image of a car”, “a photo of a car”) and kept frozen during training. Once the inputs are preprocessed, we can perform inference to get a binary segmentation map of shape (height, width) for each text query.
```
Comment (osanseviero) [inline vision_language_pretraining.md:278]: Sometimes we write preprocess and sometimes pre-process. Let's try to be consistent please

File: vision_language_pretraining.md [hunk 25]
```diff
@@ -0,0 +1,326 @@
...
+## Conclusion
+
+There have been incredible advances in multimodal models in recent years, with vision-language models making the biggest leap both in terms of performance and the variety of use cases and applications. In this blog, we talked about the latest advancements in vision-language models, as well as what multimodal datasets are available and which pre-training strategies we can use to train and fine-tune such models. We also showed how these models are integrated into 🤗 Transformers and how you can use them to perform various tasks with a few lines of code.
```
Comment (osanseviero) [inline vision_language_pretraining.md:321]:
```suggestion
There have been incredible advances in multimodal models in recent years, with vision-language models making the most significant leap in performance and the variety of use cases and applications. In this blog, we talked about the latest advancements in vision-language models, as well as what multimodal datasets are available and which pre-training strategies we can use to train and fine-tune such models. We also showed how these models are integrated into 🤗 Transformers and how you can use them to perform various tasks with a few lines of code. 
```

File: vision_language_pretraining.md [hunk 26]
```diff
@@ -0,0 +1,326 @@
...
+
+We are continuing to integrate the most impactful computer vision and multimodal models and would love to hear back from you. To stay up to date with the latest news in multimodal research, you can follow us on Twitter: [@adirik](https://twitter.com/https://twitter.com/alaradirik), [@NielsRogge](https://twitter.com/NielsRogge), [@apsdehal](https://twitter.com/apsdehal), [@a_e_roberts](https://twitter.com/a_e_roberts), [@RisingSayak](https://mobile.twitter.com/a_e_roberts), and [@huggingface](https://twitter.com/huggingface).
+
+*Acknowledgements: We thank Amanpreet Singh and Amy Roberts for their rigorous reviews. Also, thanks to Niels Rogge, Younes Belkada, Suraj Patil among many others at Hugging Face, who laid out the foundations for increasing the use of multimodal models from Transformers.*
```
Comment (osanseviero) [inline vision_language_pretraining.md:325]:
```suggestion
*Acknowledgements: We thank Amanpreet Singh and Amy Roberts for their rigorous reviews. Also, thanks to Niels Rogge, Younes Belkada, and Suraj Patil, among many others at Hugging Face, who laid out the foundations for increasing the use of multimodal models from Transformers.*
```

File: vision_language_pretraining.md [hunk 27]
```diff
@@ -0,0 +1,326 @@
...
+    2. [PrefixLM](###2-prefixlm)
+    3. [Multimodal Fusing with Cross Attention](###3-multimodal-fusing-with-cross-attention)
+    4. [MLM / ITM](###4-masked-language-modeling-image-text-matching)
+    5. [No Training](###5-no-training)
+3. [Vision-Language Models: Datasets](#vision-language-models-datasets)
+4. [Supporting Vision-Language Models in 🤗 Transformers](#supporting-vision-language-models-in-🤗-transformers)
+5. [Emerging Areas of Research](#emerging-areas-of-research)
+6. [Conclusion](#conclusion)
+
+## Vision-Language Models: Introduction
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply vision_language_pretraining.md:53]: Good one actually. We will just keep it in "Supporting Vision-Language Models in 🤗 Transformers".

**Comment by sayakpaul** [comment]
@younesbelkada thank you ^_^

> BLIP also performs ITM, but not sure about its training strategy so I am not sure if it's relevant to add it in 4) Masked-Language Modeling / Image-Text Matching wdyt?

Since BLIP has additional objectives too, we didn't include it. 

@osanseviero thanks for your comments and suggestions. Done addressing them all.

**Comment by alaradirik** [comment]
> @alaradirik I had to update an entry for the blog in `_blog.yml` to preview using Codespaces.

> **Update** Both have been fixed now.

Thank you @sayakpaul, I added BridgeTower and fixed the merge conflict, I will merge the PR on Friday.

Also thanks to @osanseviero and @younesbelkada for the reviews!

**Pull request merged**